src/rbd_fuse/rbd-fuse.cc

```diff
@@ -877,13 +865,59 @@ connect_to_cluster(rados_t *pcluster)
 	return r;
 }
 
+int init_cluster(int argc, char **argv, char **remargv)
+{
+	int r, remargc;
+
+	r = rados_create(&cluster, NULL);
+	if (r < 0) {
+		simple_err("Could not create cluster handle", r);
+		exit(1);
+	}
+
+	r = rados_conf_parse_argv_remainder(cluster, argc, (const char **)
+						argv, (const char **) remargv);
+	if (r < 0) {
+		simple_err("Error parsing the arguments", r);
+		exit(1);
+	}
+
+	for (remargc = 0; remargv[remargc] != NULL and remargc < argc;
```

Comment:
I think you need to check for `remargc < argc` before trying `remargv[remargc]`, i.e. swap these predicated, otherwise you are reading out of the buffer when `remargc == argc`.